clients/spark/core/src/main/scala/io/treeverse/gc/DataLister.scala

```diff
@@ -11,21 +11,20 @@ import org.apache.hadoop.fs.FileStatus
 
 /** List all the files under a given path.
  */
-trait DataLister {
-  @transient lazy val spark = SparkSession.active
-
-  def listData(configMapper: ConfigMapper, path: Path): DataFrame
+abstract class DataLister {
+  @transient lazy val spark: SparkSession = SparkSession.active
+  def listData(configMapper: ConfigMapper, path: Path, prefix: String): DataFrame
 }
 
 class NaiveDataLister extends DataLister {
-  override def listData(configMapper: ConfigMapper, path: Path): DataFrame = {
+  override def listData(configMapper: ConfigMapper, path: Path, prefix: String): DataFrame = {
     import spark.implicits._
     val fs = path.getFileSystem(configMapper.configuration)
     val dataIt = fs.listFiles(path, false)
     val dataList = new ListBuffer[(String, Long)]()
     while (dataIt.hasNext) {
       val fileStatus = dataIt.next()
-      dataList += ((fileStatus.getPath.toString, fileStatus.getModificationTime()))
+      dataList += ((fileStatus.getPath.getName, fileStatus.getModificationTime))
```

Comment:
What's the difference?

Reply:
toString provides the full url path, getName returns only the name